client/packages/host/src/components/Footer/StoreSelector.tsx

```diff
@@ -17,26 +19,37 @@ export const StoreSelector: FC<PropsWithChildrenOnly> = ({ children }) => {
   const t = useTranslation();
   const navigate = useNavigate();
   const { store, setStore, token } = useAuthContext();
-  const { hide, PaperClickPopover } = usePaperClickPopover();
   const { data, isLoading } = useUserDetails(token);
+  const popoverRef = useRef<PersistentPaperPopoverRef>(null);
 
   const storeSorter = (a: UserStoreNodeFragment, b: UserStoreNodeFragment) => {
     if (a.name < b.name) return -1;
     if (a.name > b.name) return 1;
     return 0;
   };
-  const stores = data?.stores?.nodes?.sort(storeSorter) || [];
-  if (!store?.name) return null;
+  const stores = useMemo(
+    () => data?.stores?.nodes?.sort(storeSorter) || [],
+    [data?.stores?.nodes]
+  );
+  const [search, setSearch] = useState('');
+
+  const filteredStores = useMemo(() => {
+    if (!search) return stores;
+    return stores.filter(s =>
+      s.name.toLowerCase().includes(search.toLowerCase())
+    );
+  }, [stores, search]);
 
+  if (!store?.name) return null;
   if (stores.length < 2) return <>{children}</>;
 
-  const storeButtons = stores.map(s => (
+  const storeButtons = filteredStores.map(s => (
     <FlatButton
       label={s.name + (s.isDisabled ? ` (${t('label.on-hold')})` : '')}
       disabled={s.id === store.id || !!s.isDisabled}
       onClick={async () => {
         await setStore(s);
-        hide();
+        popoverRef.current?.hide();
```

Comment:
This wasn't originally working with the PaperClickPopover component.. using ref to close it now since the hide() function isn't being exposed in the child component